examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+                    // Decode the message before parsing it
+                    string decodedMessage = Encoding.UTF8.GetString(Convert.FromBase64String(message.MessageText));
+                    try
+                    {
+                        JObject eventObject = JObject.Parse(decodedMessage);
+                        string topic = eventObject.Property("topic")?.Value.ToString();
+                        if (topic != null && topic == Environment.GetEnvironmentVariable("PrimaryStoreResourceId"))
+                        {
+                            JObject dataObject = JObject.Parse(eventObject.Property("data")?.Value.ToString());
+                            string key = dataObject["key"].ToString();
+                            string label = dataObject["label"]?.ToString();
+                            updatedKeyLabels.Add(new KeyLabel(key, label));
+                        }
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // If its not a valid JSON, ignore the message 
+                        log.LogInformation($"Queue message in invalid JSON format will be ignored.\nMessage: {decodedMessage}");
+                    }
+                }
+                catch (FormatException)
+                {
+                    // If its not in valid base64 format, ignore the message 
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static void BackupKeyValues(HashSet<KeyLabel> updatedKeyLabels, ILogger log)
+        {
+            // Connect to Azure App Configuration stores
+            ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+            ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+
+            // Read all settings from Primary Store and update secondary store
+            foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettings(new SettingSelector()))
+            {
+                KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                if(updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                {
+                    // current setting retrieved from primary store needs to be updated in secondary store
+                    secondaryAppConfigClient.SetConfigurationSetting(setting);
+                    log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+                    updatedKeyLabels.Remove(primaryStoreKeyLabel);
```

Comment:
One optimization we can add here is that if `updatedKeyLabels` becomes empty at this point, we can `break` the loop.

Reply:
done